doc/check.texi

```diff
@@ -1032,6 +1032,73 @@ respectively.
 Similar to @code{ck_assert_int_*}, but compares two unsigned integer values
 (@code{uintmax_t}) instead.
 
+@item ck_assert_floating_eq
+@itemx ck_assert_floating_ne
+@itemx ck_assert_floating_lt
+@itemx ck_assert_floating_le
+@itemx ck_assert_floating_gt
+@itemx ck_assert_floating_ge
+
+Compares two floating point numbers (@code{float}, @code{double},
+@code{long double}) and displays predefined message with condition and
+values of both input parameters on failure.  The operator used for comparison
+is different for each function and is indicated by the last two letters
+of the function name.  The abbreviations @code{eq}, @code{ne}, @code{lt},
+@code{le}, @code{gt}, and @code{ge} correspond to @code{==}, @code{!=},
+@code{<}, @code{<=}, @code{>}, and @code{>=} respectively. Beware
+using those operators for floating point numbers because of precision
+possible loss on every operation on floating point numbers. For example
+(1/3)*3==1 would return false, because 1/3==1.333... (or 1.(3) notation
+in Europe) and cannot be represented by computer logic. As another example
+1.1f in fact could be 1.10000002384185791015625 and 2.1f could be
+2.099999904632568359375 because of binary representation of floating
+point numbers.
+If you have different mathematical operations used on floating point numbers
+consider using precision comparisons or integer numbers instead. But in some
+cases those operators could be used. For example if you cyclically increment
+your floating point number only by positive or only by negative values than
+you may use @code{<}, @code{<=}, @code{>} and @code{>=} operators in tests.
+If your computations must end up with a certain value than @code{==} and
+@code{!=} operators may be used.
+
+@item ck_assert_floating_eq_tol
+@itemx ck_assert_floating_ne_tol
+@itemx ck_assert_floating_le_tol
+@itemx ck_assert_floating_ge_tol
+
+Compares two floating point numbers (@code{float}, @code{double},
+@code{long double}) with specified user tolerance set by the third
+parameter (@code{unsigned char}) and displays predefined message with condition
+and values of both input parameters and precision on failure. The abbreviations
+@code{eq}, @code{ne}, @code{le}, and @code{ge} correspond to @code{==},
+@code{!=}, @code{<=}, and @code{>=} respectively with acceptable
+error (tolerance) specified by the last parameter.
+Beware of using those operators for floating operators because of unclear bounds
```

Comment:
I mean that 0.1 is not exactly 0.1. There would be digits after some zeroes in decimal places. So in fact number would be a little greater than 10.